db/util.go

```diff
@@ -35,11 +35,26 @@ func cpIncr(bz []byte) (ret []byte) {
 			ret[i] = byte(0x00)
 		}
 	}
-	return EndingKey()
+	return nil
 }
 
-func IsKeyInDomain(key, start, end []byte) bool {
-	leftCondition := bytes.Equal(start, BeginningKey()) || bytes.Compare(key, start) >= 0
-	rightCondition := bytes.Equal(end, EndingKey()) || bytes.Compare(key, end) < 0
-	return leftCondition && rightCondition
+// See DB interface documentation for more information.
+func IsKeyInDomain(key, start, end []byte, isReverse bool) bool {
+	if !isReverse {
+		if bytes.Compare(key, start) < 0 {
```

Comment:
shouldnt we have `start != nil` here too ?

Reply:
They're the same: https://play.golang.org/p/prE7JB_tTX